connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -230,16 +231,17 @@ public static ConfigInfos generateResult(String connType, Map<String, ConfigKey>
             configValueMap.put(configName, configValue);
             if (!configKeys.containsKey(configName)) {
                 configValue.addErrorMessage("Configuration is not defined: " + configName);
-                configInfoList.add(new ConfigInfo(null, convertConfigValue(configValue)));
+                configInfoList.add(new ConfigInfo(null, convertConfigValue(configValue, null)));
             }
         }
 
         for (String configName: configKeys.keySet()) {
             ConfigKeyInfo configKeyInfo = convertConfigKey(configKeys.get(configName));
+            Type type = configKeys.get(configName).type;
```

Comment:
The key is also used in another dictionary map and is used much more than the value. I think using the key simplifies the code a bit.